### PR TITLE
getScaleType performance improvement

### PR DESCRIPTION
**Pull request by EricSmekens**
If you have feedback (always welcome) then I can add a commit to this PR as well.

EDIT: woops, I branched too late. I guess that will be fixed when that PR gets in, otherwise I rebranch this.

-------------------------------------------

**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
It walks over the whole array twice.

**What is the new behavior?**

> .every() method executes the provided callback function once for each element present in the array until it finds one where callback returns a falsy value.

Will be much faster, but only noticable with larger charts. I didn't do a analysis, but I guess this is an improvement anyway. (I could do some testing or set up a jsperf if you are interested.)

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/heat-map/heat-map.component.ts [hunk 1]
```diff
@@ -118,7 +119,7 @@ export class HeatMapComponent extends BaseChartComponent {
     this.yDomain = this.getYDomain();
     this.valueDomain = this.getValueDomain();
 
-    this.scaleType = this.getScaleType(this.valueDomain);
+    this.scaleType = getScaleType(this.valueDomain, true);
```
Comment (marjan-georgiev) [inline src/heat-map/heat-map.component.ts]: Should the second parameter be false? Since we're not checking for dates here.

**Comment by EricSmekens** [comment]
Totally correct, will change that this evening!

Outlook voor Android<https://aka.ms/ghei36> downloaden

________________________________

**Comment by EricSmekens** [comment]
Corrected it, thanks for feedback!

**Comment by marjan-georgiev** [comment]
Thank you! :+1:

**Pull request merged**